src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
+        }
+        Session session = this.factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Error in adding actor", e);
```

Comment:
The exception message should be specific to the operation being performed. Here, you should mention that adding the movie failed, not the actor.